### PR TITLE
No blas

### PR DESCRIPTION
**Pull request by DvirDukhan**
This PR:
1. Removes OpenBLAS and introduces native vector operations for SSE, AVX and AVX512 for both IP and L2 spaces
2. Solves a bug where `inf` is the score in brute force algorithm
3. as a result of 1+2 brute force algorithm search for top-k has been modified and without noticeable performance lost:

Before: Benchmarked simplified version of brute force algorithm via Google Bench with 1M*128 vectors of FP32
OpenBlas_top_10_ip           43784172 ns  ~ 43 ms
After: Benchmarked viae2e test of the python bindings:
lookup time for 1M *128 vectors took 0.04492020606994629 seconds ~ 44ms

### CODE REVIEW

File: src/VecSim/spaces/IP/IP_AVX.cpp [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    size_t qty16 = qty / 16;
+    size_t qty4 = qty / 4;
+
+    const float *pEnd1 = pVect1 + 16 * qty16;
+    const float *pEnd2 = pVect1 + 4 * qty4;
+
+    __m256 sum256 = _mm256_set1_ps(0);
+
+    while (pVect1 < pEnd1) {
+        //_mm_prefetch((char*)(pVect2 + 16), _MM_HINT_T0);
```
Comment (alonre24) [inline src/VecSim/spaces/IP/IP_AVX.cpp:56]: comment?

File: src/VecSim/spaces/IP/IP_AVX512.cpp [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    float *pVect2 = (float *)pVect2v;
+    size_t qty = *((size_t *)qty_ptr);
+    size_t qty16 = qty / 16;
+
+    const float *pEnd1 = pVect1 + 16 * qty16;
+
+    __m512 sum512 = _mm512_set1_ps(0);
+
+    while (pVect1 < pEnd1) {
+        //_mm_prefetch((char*)(pVect2 + 16), _MM_HINT_T0);
```
Comment (alonre24) [inline src/VecSim/spaces/IP/IP_AVX512.cpp:20]: comment?

**Review by alonre24 — COMMENTED** [review]
Looks good, added some questions. Consider adding some documentation to the vectorization functions implementation if you believe it is necessary (even though I understand the big picture, I couldn't understand the detailed implementation there...)

File: src/VecSim/spaces/space_aux.cpp [hunk 3]
```diff
@@ -0,0 +1,109 @@
+#include "space_aux.h"
+#include <cpuid.h>
+
+void cpuid(int info[4], int InfoType) {
```
[the remark this answers is not in the document]
Comment (DvirDukhan) [reply src/VecSim/spaces/space_aux.cpp:4]: extracts the cpu information

[the remark this answers is not in the document]
Comment (alonre24) [reply src/VecSim/spaces/space_aux.cpp:4]: What information? what are the flags stored in the `info` array?

**Review by alonre24 — APPROVED** [review]

**Pull request merged**